include/rewd_controllers/FTThresholdClient.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef REWD_CONTROLLERS_FTTHRESHOLDCLIENT_HPP
+#define REWD_CONTROLLERS_FTTHRESHOLDCLIENT_HPP
+
+#include <actionlib/client/simple_action_client.h>
+#include <pr_control_msgs/SetForceTorqueThresholdAction.h>
+#include <ros/ros.h>
+
+namespace rewd_controllers {
+
+/// The FTThresholdClient configures the MoveUntilTouch(Topic)Controller's
+/// thresholds.
+/// When those thresholds are exceeded, the controller stops the movement.
+class FTThresholdClient
+{
+
+public:
+  /// Constructor.
+  FTThresholdClient(const std::string& controllerThresholdTopic, ros::NodeHandle nodeHandle);
+
+  /// Sets the MoveUntilTouchControllers Thresholds.
+  /// Blocks until the threshold could be set successfully.
+  /// Can be aborted with !ros::ok(), in which case it returns false
+  bool trySetThresholdRepeatedly(double forceThreshold, double torqueThreshold);
+
+  /// Sets the MoveUntilTouchControllers Thresholds.
+  /// Throws a runtime_error if we are unable to set the thresholds.
+  void setThresholdOrThrow(double forceThreshold, double torqueThreshold, double timeout = 3.0);
```

Comment:
It is very convenient in the feeding demo code, because we call it very often ;)
I definitely want to have this function, but it's no problem to move it to the feeding demo code, if you prefer that.

Reply:
Hmm, which exceptions are you trying to raise here? I think if you're looking to catch bad threshold values, the updated `setThresholds` should continue to handle that. If you're trying to catch timeouts and retry, then the retry function could be used for that (if we observe that it's still necessary).

Reply:
Hm. That was a good question to ask. So the desired behavior in the feeding demo is "Quit the program and output why it happened".
We can say that `setThreshold` only returns false when it times out and throws an error otherwise, that way I can still know what happened.

Reply:
That sounds good to me!